### PR TITLE
Add batch rule for `native_dropout_backward`

### PR DESCRIPTION
**Pull request by guilhermeleobas**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #140140

Fixes: #122432

cc @zou3519 @Chillee @samdow @kshitij12345

### CODE REVIEW

File: aten/src/ATen/functorch/BatchRulesRandomness.cpp [hunk 1]
```diff
@@ -462,6 +467,7 @@ TORCH_LIBRARY_IMPL(aten, FuncTorchVmapMode, m) {
   UNARY_POINTWISE_RANDOM_LEADING_FLOAT(normal, float_Tensor);
 
   m.impl("native_dropout", native_dropout_batching_rule); // needs special casing because cuda version doesn't call bernoulli
+  m.impl("native_dropout_backward", native_dropout_backward_batch_rule);
```
Comment (guilhermeleobas) [inline aten/src/ATen/functorch/BatchRulesRandomness.cpp:470]: This should be `OP_DECOMPOSE`.

**Review by zou3519 — APPROVED** [review]
looks fine, just move to BatchRulesDecomposition please

**Comment by guilhermeleobas** [comment]
Hit an error for `CompositeImplicitAutograd`:

```
AssertionError: The registrations in BatchedDecompositions.cpp must be for CompositeImplicitAutograd operations. If your operation aten::native_dropout_backward is not CompositeImplicitAutograd, then please register it to the FuncTorchBatched key in another file.
```

**Pull request closed without merging**